_posts/2025-03-13-CPANSec-Retrospective-2024.md

```diff
@@ -0,0 +1,146 @@
+
+## Working on "TLS in core"
+A vanilla install of Perl (distribution) does not provide HTTPs capabilities, and this is annoying.
+That's an important (and difficult) topic that is addressed, across core developpers and CPANSec.
+
+Things started to go into motion in 2024 to make it happen.
+
+[Discussions](https://www.nntp.perl.org/group/perl.perl5.porters/2024/11/msg269050.html), with the Perl Steering Council or core contributor. A [plan](https://www.nntp.perl.org/group/perl.perl5.porters/2024/11/msg269202.html) was built, with some preliminary work happening.
+
+The creation of the new module [Crypt::Bear](https://metacpan.org/dist/Crypt-Bear) to wrap the TLS library BearSSL.
+
+## SBOM
+CPANSec attended a billion of meetings, cross connected with several other organization working on the topic. 
+
+One of the outcome of this huge effort is available as a document [Roles and metadata in open source supply-chains](https://security.metacpan.org/docs/supplychain-sbom.html) but it goes well beyond that.
+
+## Adopting security related modules
+CPANSec adopted security related modules:
+- [Crypt::DSA](https://metacpan.org/dist/Crypt-DSA)
+- [Crypt::DES_EDE3](https://metacpan.org/dist/Crypt-DES_EDE3)
+- [Crypt::OpenPGP](https://metacpan.org/dist/Crypt-OpenPGP)
+- [Crypt::OpenSSL::AES](https://metacpan.org/dist/Crypt-OpenSSL-AES)
+- [Crypt::OpenSSL::Blowfish](https://metacpan.org/dist/Crypt-OpenSSL-Blowfish)
+- [Crypt::OpenSSL::ConfiguredAPI](https://metacpan.org/dist/Crypt-OpenSSL-ConfiguredAPI)
+- [Crypt::OpenSSL::PKCS10](https://metacpan.org/dist/Crypt-OpenSSL-PKCS10)
+- [Crypt::OpenSSL::SignCSR](https://metacpan.org/dist/Crypt-OpenSSL-SignCSR)
+- [Crypt::OpenSSL::Verify](https://metacpan.org/dist/Crypt-OpenSSL-Verify)
+- [Crypt::OpenSSL::VerifyX509](https://metacpan.org/dist/Crypt-OpenSSL-VerifyX509)
+- [Crypt::Primes](https://metacpan.org/dist/Crypt-Primes)
+- [Crypt::Random](https://metacpan.org/dist/Crypt-Random)
+- [Module::Signature](https://metacpan.org/dist/Module-Signature)
+  
+Adopting modules is not always the only option and CPANSec also reviewed modules for unsecure algorithms
+in order to make recommendations or push some modules to deprecation.
+
+## Reviewing CPAN for supply chain attacks
+Through the year, CPANSec tried to come with analysis of the CPAN ecosystem 
+to know exactly how much vulnerable it would be to some common supply chain attacks:
+
+- [Study CPAN Dependency Confusion](https://security.metacpan.org/docs/cpan-dependency-confusion.html)
+- [Study CPAN StarJacking](https://security.metacpan.org/docs/cpan-starjacking.html)
+
+## Pentesting PAUSE
+CPANSec reviewed and tested PAUSE with various attempts to break it with nasty modules, steal namespaces or execute remote code.
+
+This was done on a local instance that was a good opportunity to work on automating more the reproducibility and ease of setup.
```

Comment:
…on a local instance made possible by the hard work from PEOPLE. This gave us also a good opportunity to work on …

Reply:
Done